mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -126,12 +131,16 @@ public MailboxCounters getMailboxCounters(Mailbox mailbox) throws MailboxExcepti
     public void delete(Mailbox mailbox, MailboxMessage message) {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
 
-        messageIdDAO.retrieve(mailboxId, message.getUid())
-            .thenCompose(optional -> optional.map(this::deleteUsingMailboxId)
-                .orElse(CompletableFuture.completedFuture(null)))
+        deleteAsFuture(message, mailboxId)
             .join();
     }
 
+    private CompletableFuture<Void> deleteAsFuture(MailboxMessage message, CassandraId mailboxId) {
+        return messageIdDAO.retrieve(mailboxId, message.getUid())
+            .thenCompose(optional -> optional.map(this::deleteUsingMailboxId)
```

Comment:
I would put : 

```
.map(this::deleteUsingMailboxId)
```
and 
```
.orElse(CompletableFuture.completedFuture(null)))
```
on the same ident level to make it blattant it' one or the other one...